funding/manager.go

```diff
@@ -3108,34 +3109,130 @@ func (f *Manager) waitForFundingConfirmation(
 	log.Infof("Waiting for funding tx (%v) to reach %v confirmations",
 		txid, numConfs)
 
-	var confDetails *chainntnfs.TxConfirmation
-	var ok bool
+	// setConfirmationHeight is a helper that marks the confirmation height
+	// of the channel if it hasn't been set yet. It returns false if an
+	// error occurs.
+	setConfirmationHeight := func(height uint32) bool {
+		if completeChan.ConfirmationHeight == 0 {
+			err := completeChan.MarkConfirmationHeight(height)
+			if err != nil {
+				log.Errorf("failed to update confirmed state "+
+					"for ChannelPoint(%v): %v",
+					completeChan.FundingOutpoint, err)
+
+				return false
+			}
+		}
+
+		return true
+	}
 
 	// Wait until the specified number of confirmations has been reached,
 	// we get a cancel signal, or the wallet signals a shutdown.
-	select {
-	case confDetails, ok = <-confNtfn.Confirmed:
-		// fallthrough
+	for {
+		select {
+		case updDetails, ok := <-confNtfn.Updates:
+			if !ok {
+				log.Warnf("ChainNotifier shutting down, "+
+					"cannot process updates for "+
+					"ChannelPoint(%v)",
+					completeChan.FundingOutpoint)
 
-	case <-cancelChan:
-		log.Warnf("canceled waiting for funding confirmation, "+
-			"stopping funding flow for ChannelPoint(%v)",
-			completeChan.FundingOutpoint)
-		return
+				return
+			}
 
-	case <-f.quit:
-		log.Warnf("fundingManager shutting down, stopping funding "+
-			"flow for ChannelPoint(%v)",
-			completeChan.FundingOutpoint)
-		return
-	}
+			log.Debugf("funding tx %s received confirmation in "+
```

Comment:
can you here in this log line also mention how many confirmations we still need to wait `NumConsLeft` otherwise it can be misleading.